src/main/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/codecs/JSONCodec.java

```diff
@@ -28,42 +29,71 @@
  *     CQL expects.
  * @param targetCQLType {@link DataType} of the CQL column type the Java object needs to be
  *     transformed into.
- * @param fromJava Function that transforms the Java object into the CQL object
+ * @param toCQL Function that transforms the Java object into the CQL object
  * @param <JavaT> The type of the Java object that needs to be transformed into the type CQL expects
  * @param <CqlT> The type Java object the CQL driver expects
  */
 public record JSONCodec<JavaT, CqlT>(
-    GenericType<JavaT> javaType, DataType targetCQLType, FromJava<JavaT, CqlT> fromJava)
-    implements BiPredicate<DataType, Object> {
+    GenericType<JavaT> javaType,
```

Comment:
The codec looks fine for primitive type. Needs a revisit when we doing complex types where only few fields will need to be returned. Will we be creating custom Codec based on user requests?

Reply:
added a TODO